src/coreclr/src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -228,39 +229,50 @@ private static CustomAttributeType InitCustomAttributeType(RuntimeType parameter
 
             return new CustomAttributeType(encodedType, encodedArrayType, encodedEnumType, enumName);
         }
+
         private static IList<CustomAttributeData> GetCustomAttributes(RuntimeModule module, int tkTarget)
         {
-            CustomAttributeRecord[] records = GetCustomAttributeRecords(module, tkTarget);
+            MetadataImport scope = module.MetadataImport;
+            scope.EnumCustomAttributes(tkTarget, out MetadataEnumResult attributeTokens);
+            if (attributeTokens.Length == 0) return Array.Empty<CustomAttributeData>();
+
+            CustomAttributeRecord[]? array = null;
+            if (attributeTokens.Length > MetadataEnumResult.SmallSizeLength)
+            {
+                array = ArrayPool<CustomAttributeRecord>.Shared.Rent(attributeTokens.Length);
```

Comment:
Is the large number of custom attributes common enough for ArrayPool to make a difference? The ArrayPool is not exactly free either...

Reply:
Probably not